helper/Data.php

```diff
@@ -117,7 +124,7 @@ public function fetchPaymentMethods(Cart $cart, $language)
             return array();
         }
 
-        if (!\Validate::isLoadedObject($customer)) {
+        if (!\Validate::isLoadedObject($customer) && !$this->areGuestAndOnePageCheckout16()) {
```

Comment:
not sure if I get this correctly.
Isn't the customer object required for both 1.6 and 1.7?